features/machine/machine.feature

```diff
@@ -0,0 +1,94 @@
+Feature: Machine features testing
+
+  # @author jhou@redhat.com
+  # @case_id OCP-21196
+  @smoke
+  Scenario: Machines should be linked to nodes
+    Given I have an IPI deployment
+    Then the machines should be linked to nodes
+
+  # @author jhou@redhat.com
+  # @case_id OCP-22115
+  @smoke
+  Scenario: machine-api clusteroperator should be in Available state
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Available')` is stored in the :co_available clipboard
+    Then the expression should be true> cb.co_available["status"]=="True"
+
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Degraded')` is stored in the :co_degraded clipboard
+    Then the expression should be true> cb.co_degraded["status"]=="False"
+
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Upgradeable')` is stored in the :co_upgradable clipboard
+    Then the expression should be true> cb.co_upgradable["status"]=="True"
+
+    Given evaluation of `cluster_operator('machine-api').condition(type: 'Progressing')` is stored in the :co_progressing clipboard
+    Then the expression should be true> cb.co_progressing["status"]=="False"
+
+  # @author jhou@redhat.com
+  # @case_id OCP-25436
+  @admin
+  @destructive
+  Scenario: Scale up and scale down a machineSet
+    Given I have an IPI deployment
+    And I store all machinesets to the :machinesets clipboard
+
+    Given evaluation of `cb.machinesets[0]` is stored in the :machineset clipboard
+    And evaluation of `cb.machineset.desired_replicas` is stored in the :replicas_original clipboard
+    And evaluation of `cb.machineset.desired_replicas.to_i + 1` is stored in the :replicas_expected clipboard
+    And I store the number of machines in the :num_machines_original clipboard
+
+    # scale up
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= cb.machineset.name %>   |
+      | replicas | <%= cb.replicas_expected %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+
+    # register clean-up just in case of failure
+    Given I register clean-up steps:
+    """
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= cb.machineset.name %>   |
+      | replicas | <%= cb.replicas_original %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+    """
+
+    # num of machines should increase
+    And I wait for the steps to pass:
+    """
+    Given I store the number of machines in the :num_machines_current clipboard
+    Then the expression should be true> cb.num_machines_current.to_i == cb.num_machines_original.to_i + 1
```

Comment:
I'm thinking that there is no need for step `I store the number of machines in the :num_machines_current clipboard`, likely this should do:
```
Then the expression should be true> machine_set.current_replicas == cb.num_machines_original.to_i + 1
```

Reply:
@akostadinov this step is to verify the total number of machines have grown. The `machine_set.current_replicas` does not reflect the total number of machines in the cluster, because there are usually multiple machinesets. 
For example, machineset A has replicas=2, machineset B and C also have replicas=2, so the total number of machines could be 6.